### PR TITLE
MAINT: Prepare for SciPy 1.4.1

### PR DESCRIPTION
**Pull request by tylerjereddy**
- release notes/version updates
- Backport of #11238

Please check carefully---working fast!

### CODE REVIEW

**Comment by rgommers** [comment]
LGTM

File: doc/release/1.4.1-notes.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+==========================
+SciPy 1.4.1 Release Notes
+==========================
+
+.. contents::
+
+SciPy 1.4.1 is a bug-fix release with no new features
+compared to 1.4.0. Importantly, it aims to fix a problem
+where an older version of pybind11 may cause a segmentation
+fault in some cases.
```
Comment (peterbell10) [inline doc/release/1.4.1-notes.rst]: How about "when imported alongside incompatible libraries"?

Comment (tylerjereddy) [reply doc/release/1.4.1-notes.rst]: Thanks Peter, I've revised accordingly.

**Comment by tylerjereddy** [comment]
Let's keep an eye on the `scipy.optimize.tests.test__differential_evolution.TestDifferentialEvolutionSolver` failure in the matrix over in wheels for v1.4.x as well:

https://travis-ci.org/MacPython/scipy-wheels/jobs/626422295

Fluke?

**Comment by charris** [comment]
>Fluke?

You should be able to rerun the build just like tests can be rerun.

**Comment by andyfaff** [comment]
Not clear why that test has failed. I don't think the DE code has changed. Has anything changed with random number generation?

**Comment by tylerjereddy** [comment]
Backport CI green --merging

**Pull request merged**